### PR TITLE
WIP: Tier metrics

### PR DESCRIPTION
**Pull request by krisis**
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
```
Error running mint automation
+ mkdir -p 12747-f801adb/gopath/src/github.com/minio
+ git -C 12747-f801adb/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 12747-f801adb/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 12747-f801adb/gopath/src/github.com/minio/minio fetch --quiet minio pull/12747/head:pr12747
+ git -C 12747-f801adb/gopath/src/github.com/minio/minio checkout --quiet pr12747
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/12747-f801adb/gopath
+ make -C 12747-f801adb/gopath/src/github.com/minio/minio --quiet
go: github.com/minio/madmin-go@v1.0.17 (replaced by ../madmin-go): reading ../madmin-go/go.mod: open /home/mint-auto/mint-auto/12747-f801adb/gopath/src/github.com/minio/madmin-go/go.mod: no such file or directory
go: github.com/minio/madmin-go@v1.0.17 (replaced by ../madmin-go): reading ../madmin-go/go.mod: open /home/mint-auto/mint-auto/12747-f801adb/gopath/src/github.com/minio/madmin-go/go.mod: no such file or directory
make: *** [build] Error 1

```

**Comment by minio-trusted** [comment]
```
Error running mint automation
+ mkdir -p 12747-3eb2034/gopath/src/github.com/minio
+ git -C 12747-3eb2034/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 12747-3eb2034/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 12747-3eb2034/gopath/src/github.com/minio/minio fetch --quiet minio pull/12747/head:pr12747
+ git -C 12747-3eb2034/gopath/src/github.com/minio/minio checkout --quiet pr12747
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/12747-3eb2034/gopath
+ make -C 12747-3eb2034/gopath/src/github.com/minio/minio --quiet
go: github.com/minio/madmin-go@v1.0.17 (replaced by ../madmin-go): reading ../madmin-go/go.mod: open /home/mint-auto/mint-auto/12747-3eb2034/gopath/src/github.com/minio/madmin-go/go.mod: no such file or directory
go: github.com/minio/madmin-go@v1.0.17 (replaced by ../madmin-go): reading ../madmin-go/go.mod: open /home/mint-auto/mint-auto/12747-3eb2034/gopath/src/github.com/minio/madmin-go/go.mod: no such file or directory
make: *** [build] Error 1

```

File: cmd/data-scanner.go [hunk 1]
```diff
@@ -830,6 +830,7 @@ type sizeSummary struct {
 	replicaSize    int64
 	pendingCount   uint64
 	failedCount    uint64
+	tierStats      TierStatsCache
```
Comment (klauspost) [inline cmd/data-scanner.go:833]: I fail to see where this is being set/updated? Is that part missing?

File: cmd/data-usage-cache.go [hunk 2]
```diff
@@ -55,6 +55,7 @@ type dataUsageEntry struct {
 	Versions         uint64 // Versions that are not delete markers.
 	ObjSizes         sizeHistogram
 	ReplicationStats *replicationStats
+	TieringStats     TierStatsCache
```
Comment (klauspost) [inline cmd/data-usage-cache.go:58]: I would like this to be a nil pointer if tiering isn't enabled, so nil means that no data is tiered.

That way non-tiered setups don't pay the memory price.

File: cmd/tier-stats.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	}
+}
+
+var globalTieringStats *NodeTieringStats
+
+//msgp:ignore NodeTieringStats
+type NodeTieringStats struct {
+	sync.RWMutex
+	UsageCache TierStatsCache
+	NodeCache  TierStatsCache
```
Comment (klauspost) [inline cmd/tier-stats.go:53]: Explain what these are.

File: cmd/tier-stats.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import (
+	"context"
+	"sync"
+)
+
+//go:generate msgp -file $GOFILE -unexported
+
+//msgp:tuple tieringStats
+type tieringStats struct {
+	TieredSize uint64 `json:"tieredSize"`
```
Comment (klauspost) [inline cmd/tier-stats.go:29]: How about a version count?

File: cmd/data-scanner.go [hunk 5]
```diff
@@ -877,7 +878,7 @@ func (i *scannerItem) applyHealing(ctx context.Context, o ObjectLayer, meta acti
 	return res.ObjectSize
 }
 
-func (i *scannerItem) applyLifecycle(ctx context.Context, o ObjectLayer, meta actionMeta) (applied bool, size int64) {
+func (i *scannerItem) applyLifecycle(ctx context.Context, o ObjectLayer, meta actionMeta, sizeS *sizeSummary) (applied bool, size int64) {
```
Comment (klauspost) [inline cmd/data-scanner.go:881]: Unused change AFAICT.

**Comment by krisis** [comment]
@klauspost I am closing this PR in favor of https://github.com/minio/minio/pull/13381.

**Pull request closed without merging**